File: mywebsite.client/src/app/api/models/cvdto.ts
/* tslint:disable */
/* eslint-disable */
import { CertificateDto } from '../models/certificate-dto';
import { ContactDto } from '../models/contact-dto';
import { EducationDto } from '../models/education-dto';
import { LanguageDto } from '../models/language-dto';
import { SkillDto } from '../models/skill-dto';
import { WorkExperienceDto } from '../models/work-experience-dto';
export interface Cvdto {
  address?: string | null;
  birthDate?: string;
  certificates?: Array<CertificateDto> | null;
  contacts?: Array<ContactDto> | null;
  description?: string | null;
  education?: Array<EducationDto> | null;
  fullName?: string | null;
  id?: string;
  isMale?: boolean;
  languages?: Array<LanguageDto> | null;
  nationality?: string | null;
  picture?: string | null;
  profession?: string | null;
  skills?: Array<SkillDto> | null;
  workExperience?: Array<WorkExperienceDto> | null;
}

File: MyWebSite.Server/Data/DTOs/CVDTO.cs
using MyWebSite.Server.Data.Entities;

namespace MyWebSite.Server.Data.DTOs
{
    public class CVDTO
    {
        public Guid Id { get; set; }
        public string Picture { get; set; }
        public string FullName { get; set; }
        public bool IsMale { get; set; }
        public string Nationality { get; set; }
        public DateTime BirthDate { get; set; }
        public string Address { get; set; }
        public string Profession { get; set; }
        public string Description { get; set; }
        public List<ContactDTO>? Contacts { get; set; }
        public List<SkillDTO>? Skills { get; set; }
        public List<WorkExperienceDTO>? WorkExperience { get; set; }
        public List<EducationDTO>? Education { get; set; }
        public List<LanguageDTO>? Languages { get; set; }
        public List<CertificateDTO>? Certificates { get; set; }
    }
}
